### PR TITLE
Add vlm generation function

### PR DESCRIPTION
**Pull request by meatybobby**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/vlm/inference/vlm_inference_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+
+class VLMInferenceWrapper(AbstractModelInferenceWrapper):
+    """Constructor for the model inference wrapper
+
+    The wrapper prepares the model for inference, provides the required input
+    data, and runs the forward pass
+
+    Args:
+        model (T5Model): The T5 model (MCore or legacy)
```
Comment (yaoyu-33) [inline nemo/collections/vlm/inference/vlm_inference_wrapper.py:36]: fix

File: nemo/collections/vlm/inference/vlm_inference_controller.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    def detokenize(self, tokens):
+        return self._tokenizer.decode(tokens, skip_special_tokens=True)
+
+    def tokenize(self, prompt):
+        return self._tokenizer.encode(prompt, add_special_tokens=False)
+
+
+class VLMTextGenerationController(SimpleTextGenerationController):
+    def __init__(self, inference_wrapped_model, processor):
```
Comment (yaoyu-33) [inline nemo/collections/vlm/inference/vlm_inference_controller.py:39]: add tokenizer, and image_processor as 2 args

**Comment by meatybobby** [comment]
@yaoyu-33 Do we need to modify [mllama_generate.py](https://github.com/NVIDIA/NeMo/blob/main/scripts/vlm/mllama_generate.py) to use new generate function?

File: nemo/collections/vlm/mllama/model/base.py [hunk 3]
```diff
@@ -467,14 +469,23 @@ def forward(
...
             )
 
+            original_position_id = position_ids[0]
+
+            if inference_params is not None:
+                inference_params.xattn_caches = xattn_caches
+                inference_params.cross_attention_masks = cross_attention_masks
+                inference_params.full_text_row_masked_out_mask = full_text_row_masked_out_mask
+        else:
+            original_position_id = [cross_attention_masks.shape[2] - 1]
```
Comment (yaoyu-33) [inline nemo/collections/vlm/mllama/model/base.py:479]: is this a single id for generation? i feel original position id name is a bit confusing

File: nemo/collections/vlm/inference/base.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+    inference_wrapped_model = setup_inference_wrapper(
+        model, tokenizer, params_dtype, inference_batch_times_seqlen_threshold
+    )
+
+    return inference_wrapped_model, processor
+
+
+def generate(
+    model: VLMInferenceWrapper,
```
Comment (yaoyu-33) [inline nemo/collections/vlm/inference/base.py:86]: rename this model to wrapped model, just to be clear it needs to be wrapped first

**Review by yaoyu-33 — APPROVED** [review]
